x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/SubTests.java

```diff
@@ -221,6 +204,29 @@ public static Iterable<Object[]> parameters() {
                 "SubUnsignedLongsEvaluator"
             )
         );
+        suppliers = anyNullIsNull(suppliers, (nullPosition, nullValueDataType, original) -> {
+            if (nullValueDataType == DataType.NULL) {
+                return original.getData().get(nullPosition == 0 ? 1 : 0).type();
+            }
+            return original.expectedType();
+        }, (nullPosition, nullData, original) -> original);
+
+        suppliers.add(new TestCaseSupplier("MV", List.of(DataType.INTEGER, DataType.INTEGER), () -> {
```

Comment:
Nit: Any reason to add this after the anyNullIsNull call? Just to preserve the typical order if possible

Reply:
`anyNullIsNull` does funny things withe the warnings we expect. I'll add a comment explaining.